hack/xref-quadlet-docs

```diff
@@ -0,0 +1,261 @@
+    ) or die "Try `$ME --help' for help\n";
+}
+
+# END   boilerplate args checking, usage messages
+###############################################################################
+
+############################## CODE BEGINS HERE ###############################
+
+# The term is "modulino".
+__PACKAGE__->main()                                     unless caller();
+
+# Main code.
+sub main {
+    # Note that we operate directly on @ARGV, not on function parameters.
+    # This is deliberate: it's because Getopt::Long only operates on @ARGV
+    # and there's no clean way to make it use @_.
+    handle_opts();                      # will set package globals
+
+    # No command-line args
+    die "$ME: Too many arguments; try $ME --help\n"                 if @ARGV;
+
+    my $errs = 0;
+    $SIG{__WARN__} = sub {
+        print STDERR "@_";
+        ++$errs;
+    };
+
+    # Assume that Go source file has Truth
+    my $true_keys = read_go($Go);
+
+    # Read md file, compare against Truth
+    crossref_doc($Doc, $true_keys);
+
+    exit $errs;
+}
+
+
+#############
+#  read_go  #  Returns list of key strings found in quadlet.go
+#############
+sub read_go {
+    my $path = shift;
+    open my $fh, '<', $path
+        or die "$ME: Cannot read $path: $!\n";;
+
+    my @found;                          # List of key strings
+    my $last_constname;                 # Most recently seen const name
+
+    while (my $line = <$fh>) {
+        # Only interested in lines of the form   KeyFoo = "Foo"
+        if ($line =~ /^\s+Key(\S+)\s+=\s+"(\S+)"/) {
+            my ($constname, $keystring) = ($1, $2);
+
+            # const name *should* be the same as the string, except (sigh)
+            # there's an inexplicable block of KeyNetworkFoo = "Foo".
+            unless ($constname eq $keystring || $constname eq "Network$keystring") {
+                # There's also, sigh again, some preexisting mismatches
+                unless ($constname =~ /Remap[GU]ID/) {
+                    warn "$ME: $path:$.: mismatched strings: Key$constname = \"$keystring\"\n";
+                }
+            }
+
+
+            # Sorting check. Go by the const name (KeyFoo, left-hand side),
+            # not the string, because of the KeyNetworkFoo problem.
+            if ($last_constname) {
+                if (lc($constname) lt lc($last_constname)) {
+                    warn "$ME: $path:$.: out-of-order variable name 'Key$constname' should precede 'Key$last_constname'\n";
+                }
+            }
+            $last_constname = $constname;
+
+            # Dup check. Assume that Go compiler will detect duplicate
+            # const names, but we have to check the strings.
+            if (grep { lc($_) eq lc($keystring) } @found) {
+                # sigh: "Driver" and "Options" also exist in KeyNetworkFoo space
+                warn "$ME: $path:$.: duplicate key string \"$keystring\"\n"
+                    unless $keystring =~ /^(Driver|Options)$/;
```

Comment:
Barf again. Just checking: is it intentional to have multiple `Driver` and `Options` definitions?

Reply:
I think that is something that should be fixed in the code.
I guess this is due the fact that we have Driver/Options for Volume and Network units for example. So it would need to be documented for each unit type. But yeah there is really no reason to define the same string const twice in the code unless we like to bloat the binary size by some bytes.

Reply:
As I wrote, the `Network` part can be removed (and then also remove the duplications)